strongbox-commons/src/main/java/org/carlspring/strongbox/booters/SystemPropertiesInitializer.java

```diff
@@ -0,0 +1,56 @@
+package org.carlspring.strongbox.booters;
+
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+/**
+ * Ensures that a handful of core system properties are set prior to starting the application.
+ *
+ * @author cbono
+ */
+class SystemPropertiesInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext>
+{
+    // All of these properties are package-protected so VisibleForTesting
```

Comment:
I have used Guava `@VisibleForTesting` in the past and I see that its available already in the libs here but I have grown to be not a fan of Guava due to its kitchen sink packaging and inter-twinings w/ Soy etc.. It has really burnt us at my company in this regard so I am hesitant to add any more references to it in my code. Also, that annotation is not much more than a visual marker so I just did that marking w/ a single comment telling such.